homeassistant/components/aladdin_connect/cover.py

```diff
@@ -35,30 +42,49 @@
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up Aladdin Connect devices yaml depreciated."""
+    _LOGGER.warning(
+        "Loading aladdin_connect via platform setup is deprecated; Please remove it from your configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Aladdin Connect platform."""
-
+    config = hass.data[DOMAIN][config_entry.entry_id]
     username: str = config[CONF_USERNAME]
     password: str = config[CONF_PASSWORD]
```

Comment:
```suggestion
    username: str = config_entry.data[CONF_USERNAME]
    password: str = config_entry.data[CONF_PASSWORD]
```

Reply:
Ah, now I get it.  Sometime you just need to explain it for an amateur :)